src/compiler/types.rs

```diff
@@ -441,6 +444,12 @@ impl From<&NodeArray> for Vec<Rc<Node>> {
     }
 }
 
+impl<'node_array> From<&'node_array NodeArray> for &'node_array [Rc<Node>] {
```

Comment:
Does Rust force you to provide the lifetime annotation here? (I'm a little surprised it doesn't infer it, but I've never tried to implement a trait for a slice.)

Reply:
Ya it does require it, it seems to not infer that the lifetime of the `&NodeArray` in the type parameter `From<&NodeArray>` is obviously the same as the lifetime of the corresponding `&NodeArray` in the method argument `node_array: &NodeArray`